zerver/management/commands/send_password_reset_email.py

```diff
@@ -50,10 +57,18 @@ def send(self, users: List[UserProfile]) -> None:
                 "realm_name": user_profile.realm.name,
                 "active_account_in_realm": True,
             }
-            send_email(
-                "zerver/emails/password_reset",
-                to_user_ids=[user_profile.id],
-                from_address=FromAddress.tokenized_no_reply_address(),
-                from_name=FromAddress.security_email_from_name(user_profile=user_profile),
-                context=context,
-            )
+
+            for i in range(0, retry_number):
+                try:
+                    send_email(
+                        "zerver/emails/password_reset",
+                        to_user_ids=[user_profile.id],
+                        from_address=FromAddress.tokenized_no_reply_address(),
+                        from_name=FromAddress.security_email_from_name(user_profile=user_profile),
+                        context=context,
+                    )
+                except EmailNotDeliveredException:
+                    if i == retry_number - 1:
+                        print(f"Failed to send to {user_profile.delivery_email}")
```

Comment:
I think we want a `traceback.print_exc()` here.  Otherwise, if none of the emails can be sent, you won't see the exception at all.

Reply:
Hmm, I'm not sure that's needed. We get traceback output to the console due the logging.exception calls:
```
    try:
        # This will call .open() for us, which is a no-op if it's already open;
        # it will only call .close() if it was not open to begin with
        if connection.send_messages([mail]) == 0:
            logger.error("Unknown error sending %s email to %s", template, mail.to)
            raise EmailNotDeliveredException
    except smtplib.SMTPResponseException as e:
        logger.exception(
            "Error sending %s email to %s with error code %s: %s",
            template,
            mail.to,
            e.smtp_code,
            e.smtp_error,
            stack_info=True,
        )
        raise EmailNotDeliveredException
    except smtplib.SMTPException as e:
        logger.exception(
            "Error sending %s email to %s: %s", template, mail.to, str(e), stack_info=True
        )
        raise EmailNotDeliveredException
```

The one case where we won't get a traceback is this part:
```
        if connection.send_messages([mail]) == 0:
            logger.error("Unknown error sending %s email to %s", template, mail.to)
            raise EmailNotDeliveredException
```

But we get the `logger.error` output to the console anyway and the traceback is not useful here.